t/mojolicious/pattern.t

```diff
@@ -250,4 +250,22 @@ is_deeply $result, {'bar' => 23}, 'right structure';
 is $pattern->render($result, 1), '/foo/23/baz', 'right result';
 ok !$pattern->match('/foo/bar/baz', 1), 'no result';
 
+# Alternate placeholder characters
+$pattern = Mojolicious::Routes::Pattern->new->placeholder_start('|')->parse('/|test');
+$result  = $pattern->match('/foo');
+is_deeply $result, {test => 'foo'}, 'right structure';
+is $pattern->render($result, 1), '/foo', 'right result';
+$pattern = Mojolicious::Routes::Pattern->new->relaxed_start('|')->parse('/|test');
+$result  = $pattern->match('/foo');
+is_deeply $result, {test => 'foo'}, 'right structure';
+is $pattern->render($result, 1), '/foo', 'right result';
+$pattern = Mojolicious::Routes::Pattern->new->wildcard_start('|')->parse('/|test');
+$result  = $pattern->match('/foo');
+is_deeply $result, {test => 'foo'}, 'right structure';
+is $pattern->render($result, 1), '/foo', 'right result';
+$pattern = Mojolicious::Routes::Pattern->new->types({num => qr/\d+/})->type_start('|')->parse('/<test|num>');
+$result  = $pattern->match('/42');
+is_deeply $result, {test => '42'}, 'right structure';
+is $pattern->render($result, 1), '/42', 'right result';
```

Comment:
Make it a subtest (we do that for all new tests).

Reply:
Done